src/test/kotlin/app/batch/DecompressionProcessorTest.kt

```diff
@@ -0,0 +1,129 @@
+@ActiveProfiles("httpDataKeyService", "awsS3")
+@SpringBootTest
+@TestPropertySource(properties = [
+    "hbase.zookeeper.quorum=hbase",
+    "aws.region=eu-west-1",
+    "s3.bucket=bucket1",
+    "s3.prefix.folder=test/output/",
+    "s3.key.regex=([A-Za-z]*\\.[A-Za-z]*\\.[0-9]{4}\\.json\\.gz)",
+    "s3.data.key.extension=\\.enc$",
+    "s3.metadata.key.extension=\\.encryption\\.json$"
+])
+class DecompressionProcessorTest {
+    @MockBean
+    private lateinit var httpClientProvider: HttpClientProvider
+
+    @Test
+    fun Should_Decompress_Correctly_When_Given_Gzipped_Stream() {
+
+        val data = "dataworksdataworksdataworksdataworksdataworksdataworksdataworks"
+        val byteArray = data.toByteArray(StandardCharsets.UTF_8)
+        val testOutputStream = ByteArrayOutputStream()
+        val gzippedStream = CompressorStreamFactory().createCompressorOutputStream(CompressorStreamFactory.GZIP,
+            testOutputStream)
+        gzippedStream.write(byteArray)
+        gzippedStream.close()
+
+        val inputStream = ByteArrayInputStream(testOutputStream.toByteArray())
+        val decryptedStream = DecryptedStream(inputStream, "test.json.gz")
+        val decompressionProcessor = DecompressionProcessor()
+        val decompressed = decompressionProcessor.process(decryptedStream) as GzipCompressorInputStream
+
+        assertEquals(IOUtils.toString(decompressed, StandardCharsets.UTF_8), data)
+        assertEquals(0, data.length.compareTo(decompressed.uncompressedCount))
+    }
+
+    @Test(expected = RuntimeException::class)
+    fun Should_Throw_Exception_When_Given_Non_Gzipped_Stream() {
+        val data = "dataworksdataworksdataworksdataworksdataworksdataworksdataworks"
+        val byteArray = data.toByteArray(StandardCharsets.UTF_8)
+        val testOutputStream = ByteArrayOutputStream()
+        val bzippedStream = CompressorStreamFactory().createCompressorOutputStream(CompressorStreamFactory.BZIP2,
+            testOutputStream)
+        bzippedStream.write(byteArray)
+        bzippedStream.close()
+
+        val inputStream = ByteArrayInputStream(testOutputStream.toByteArray())
+        val decryptedStream = DecryptedStream(inputStream, "test.json.bz2")
+        val decompressionProcessor = DecompressionProcessor()
+        decompressionProcessor.process(decryptedStream)
+    }
+
+    @Test
+    fun Should__Log_Compressed_Count_When_Given_Gzipped_Stream() {
+        val root = LoggerFactory.getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME) as ch.qos.logback.classic.Logger
+        val mockAppender: Appender<ILoggingEvent> = mock()
+        root.addAppender(mockAppender)
+
+        val data = "dataworksdataworksdataworksdataworksdataworksdataworksdataworks"
+        val byteArray = data.toByteArray(StandardCharsets.UTF_8)
+        val testOutputStream = ByteArrayOutputStream()
+        val bzippedStream = CompressorStreamFactory().createCompressorOutputStream(CompressorStreamFactory.GZIP,
```

Comment:
Should be `gzippedStream`
Also, since this block is reused - worth pulling out to a function / `@Before` statement?

Reply:
Thanks Ben , Moved common code to a function